xbmc/FileItem.cpp

```diff
@@ -1066,6 +1066,9 @@ bool CFileItem::IsPythonScript() const
 
 bool CFileItem::IsType(const char *ext) const
 {
+  if (!m_strDynPath.empty())
+    return URIUtils::HasExtension(m_strDynPath, ext);
+
   return URIUtils::HasExtension(m_strPath, ext);
```

Comment:
This could have be done a bit more simple: 

<pre>
- return URIUtils::HasExtension(m_strPath, ext);
+ return URIUtils::HasExtension(GetDynPath(), ext);
</pre>

 would also do the trick as `CFileItem::getDynPath` returns `m_strDynPath` if not empty, `m_strPath`otherwise.